api/http/api_inventory.go

```diff
@@ -507,11 +507,22 @@ func (i *inventoryHandlers) updateDeviceAttributes(
 
 	// upsert or replace the attributes
 	if r.Method == http.MethodPatch {
+		// we only check if the inventory needs update with PATCH, we assume that calling PUT
+		// is a conscious choice of the caller, to write to the db unconditionally.
```

Comment:
@alfrunes heads up here: shall we keep it only in the PATCH or also check in PUT where we replace all the attributes (I am having some doubts about introducing the check in the latter case as well) (if we want this, `InventoryNeedsUpdate` will need a small update)